internal/mode/static/nginx/runtime/manager_test.go

```diff
@@ -23,6 +23,42 @@ var _ = Describe("NGINX Runtime Manager", func() {
 	})
 })
 
+func TestEnsureNginxRunning(t *testing.T) {
+	ctx := context.Background()
+	cancellingCtx, cancel := context.WithCancel(ctx)
+	time.AfterFunc(1*time.Millisecond, cancel)
+	tests := []struct {
+		ctx         context.Context
+		name        string
+		expectError bool
+	}{
+		{
+			ctx:         ctx,
+			name:        "context exceeded",
+			expectError: true,
+		},
+		{
+			ctx:         cancellingCtx,
+			name:        "context cancelled",
+			expectError: true,
+		},
```

Comment:
These two cases are the same I believe. When the context is exceeded, it's cancelled.

Let's also add the success case, when error is not expected.

Reply:
seems so, good thing to know, actually I have avoided to add success test for `EnsureNginxRunning`, and the reason seems to be I had some issues with proper mocking, I will try to go into details, so 

EnsureNginxRunning(ctx) seems to call inside findMainProcess(ctx, checkFile, readFile, timeout)

theres already a test for `findMainProcess` and when mocking these args, they are passed into fnction without problems, 

but when testing `EnsureNginxRunning` I have no idea how to properly pass these args into findMainProcess from EnsureNginxRunning, its actually even using these:

```
const (
	pidFile            = "/var/run/nginx/nginx.pid"
	pidFileTimeout     = 10000 * time.Millisecond
)
```

on first I thought, changing const to var, so I can update the actual mock of pidFile, but I suppose const is protecting this value, so I also checked mocks provided in findMainProcess, but still, after creating these mocks, 

how do I add properly, or properly apply them in the test, since EnsureNginxRunning(ctx) does not send them to `findMainProcess`, I was not sure what's the best thing even tho I tried a few examples,

any advice is appreciated a lot !!

Reply:
Ah, I see what you're saying. Since we do already test `findMainProcess`, let's not add anything else here.